### PR TITLE
drivers/pipe: Remove pipe from file system after open 

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
to make the pipe as an anonymous object as soon as possible and simplify the life cycle management.

- drivers/pipe: Add g_ prefix to pipe_fops and fifo_fops

## Impact
more conform to spec:
https://www.gnu.org/software/libc/manual/html_node/Pipes-and-FIFOs.html

## Testing
adb+libuv

### CODE REVIEW

File: drivers/pipes/pipe.c [hunk 1]
```diff
@@ -68,14 +62,10 @@ static const struct file_operations pipe_fops =
...
-#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
-  , pipecommon_unlink  /* unlink */
-#endif
 };
 
-static sem_t  g_pipesem       = SEM_INITIALIZER(1);
-static uint32_t g_pipeset     = 0;
-static uint32_t g_pipecreated = 0;
+static sem_t g_pipesem = SEM_INITIALIZER(1);
+static int   g_pipeno  = 0;
```
Comment (pkarashchenko) [inline drivers/pipes/pipe.c:68]:
```suggestion
static int   g_pipeno;
```

Comment (xiaoxiang781216) [reply drivers/pipes/pipe.c]: Done.

File: drivers/pipes/pipe.c [hunk 2]
```diff
@@ -155,69 +131,34 @@ static int pipe_register(size_t bufsize, int flags,
...
   /* Create a pathname to the pipe device */
 
   snprintf(devname, namesize, CONFIG_DEV_PIPE_VFS_PATH"/%d", pipeno);
 
-  /* Check if the pipe device has already been created */
+  /* Allocate and initialize a new device structure instance */
 
-  if ((g_pipecreated & (1 << pipeno)) == 0)
+  dev = pipecommon_allocdev(bufsize);
+  if (!dev)
```
Comment (pkarashchenko) [inline drivers/pipes/pipe.c:149]:
```suggestion
  if (dev = NULL)
```

Comment (xiaoxiang781216) [reply drivers/pipes/pipe.c]: Done.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

I was only thinking about g_pipesem usage and a critical section as alternative for atomic increment

**Comment by xiaoxiang781216** [comment]
Let's do in the next PR.

**Pull request merged**